File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
    public int solution(int n, int k) {
        int answer = -1;

        String str = Integer.toString(n, k);

        int idx = 0;

        ArrayList<String> answers = new ArrayList<>();

        StringBuilder sb = new StringBuilder();

        for (idx = 0 ; idx < str.length() ; idx++) {
            char ch = str.charAt(idx);
            if (ch != '0') {
                sb.append(ch);
            } else {
                if (sb.length() != 0) {
                    String tmp = sb.toString();
                    if (isPrime(Integer.parseInt(tmp))) {
                        answers.add(tmp);
                    }
                    sb = new StringBuilder();
                }
            }
        }

        if (sb.length() != 0) {
            String tmp = sb.toString();
            if (isPrime(Long.parseLong(tmp))) {
                answers.add(tmp);
            }
        }

        return answers.size();
    }

    public boolean isPrime(long number) {
        if (number == 1) return false;

        if (number == 2 || number == 3)
            return true;

        if (number % 2 == 0) return false;

        for (int div = 3 ; div <= Math.sqrt(number) ; div += 2) {
            if (number % div == 0) return false;
        }

        return true;
    }
}

File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math


def convert(number, base):
    tmp = ''
    while number:
        tmp = str(number % base) + tmp
        number //= base
    return tmp


def isPrime(number):
    if number == 1:
        return False
    for i in range(2, int(math.sqrt(number) + 1)):
        if number % i == 0:
            return False
    return True


def solution(n, k):
    answer = 0

    number = convert(n, k)

    tmp = ''
    for i in number:
        if i == '0':
            if len(tmp) > 0:
                if isPrime(int(tmp)):
                    answer += 1
            tmp = ''
        else:
            tmp += i

    if len(tmp) > 0:
        if isPrime(int(tmp)):
            answer += 1

    return answer